### PR TITLE
also add the sites name as a WP CLI alias

### PR DESCRIPTION
**Pull request by tomjn**
This should enable commands such as `wp @wordpress-one core version`

### CODE REVIEW

**Review by Mte90 — APPROVED** [review]

**Comment by janw-me** [comment]
```
@vvv:
  ssh: vagrant@testone.test
  path: /srv/www/testone/public_html
```

This prompts a password (at least for me), 
By default wp-cli will accept `ssh: vagrant`.
The `@testone` should not be needed.

Also see [here](https://make.wordpress.org/cli/handbook/guides/running-commands-remotely/#using-an-ssh-connection):

> The scheme argument defaults to ssh and alternately accepts options for vagrant, docker and docker-compose.

And the [source code](https://github.com/wp-cli/wp-cli/blob/1512cd07af5ffe0cdbcf563f368e182375618181/php/utils.php#L1013) that does this magic

You can also see the difference when adding the `--debug` flag and check the _Running SSH command_ output.

**Comment by Mte90** [comment]
I don't understand.
Yes it asks the password and I didn't find a way to send it with wpcli to avoid asking.

**Comment by tomjn** [comment]
I haven’t encountered this, do you have an SSH key set up?

**Comment by janw-me** [comment]
```
@vvv:
  ssh: vagrant
  path: /srv/www/testone/public_html
```

Changing it to this it will not ask for a password, I've been using this for years now, I've always added this to `wp-cli.yml` manually.
I do have an SSH key in my agent, (ubuntu host machine) and I do use my ssh key to log in to other servers.
But I've never done any tweak with vvv and ssh keys. I just install `vagrant up --provision` and after that wp cli works, and I enter ssh with `vagrant ssh`

**Comment by Mte90** [comment]
Uhm this on OSX or Linux host? 

I tested on linux and yes it is not asking a password without the domain.
If you can confirm where you tested I think that I can do a pr right now.

**Comment by janw-me** [comment]
My laptop runs Ubuntu 20.10
I've never tested this on an other OS

**Comment by tomjn** [comment]
@janw-me VVV does not add the `vagrant` domain,  whatever you're doing it should not work unless you've manually added a hosts file entry for `vagrant`

**Comment by tomjn** [comment]
Either that or you added an `ssh-config` for VVV under the alias `vagrant`, `ssh vagrant` should result in a DNS error.

This is MacOS:

```
❯ ssh vagrant
ssh: Could not resolve hostname vagrant: nodename nor servname provided, or not known
```

Also, the VM should not be asking for a password unless something has been misconfigured.

I also tested this out on MacOS and could not reproduce the problem:

```sh
❯ wp @wordpress-two core version
5.6
```

I added a `@test` alias that uses `vagrant`:

```yaml
# auto-generated file
path: "public_html"
@test:
  ssh: vagrant
  path: /srv/www/wordpress-two/public_html
@vvv:
  ssh: vagrant@two.wordpress.test
  path: /srv/www/wordpress-two/public_html
@wordpress-two:
  ssh: vagrant@two.wordpress.test
  path: /srv/www/wordpress-two/public_html
```

Then ran with every alias:

```sh
~/dev/vvv/www/wordpress-two ᚴ:master 
❯ wp post list
Warning: mysqli_real_connect(): (HY000/2002): No such file or directory in /Users/tarendai/dev/vvv/www/wordpress-two/public_html/wp-includes/wp-db.php on line 1653
Error: `No such file or directory`
Error establishing a database connection
This either means that the username and password information in your `wp-config.php` file is incorrect or we can’t contact the database server at `localhost`. This could mean your host’s database server is down.

Are you sure you have the correct username and password?
Are you sure you have typed the correct hostname?
Are you sure the database server is running?

If you’re unsure what these terms mean you should probably contact your host. If you still need help you can always visit the WordPress Support Forums. `No such file or directory`
~/dev/vvv/www/wordpress-two ᚴ:master 
❯ wp @test post list
Error: Cannot connect over SSH using provided configuration.
~/dev/vvv/www/wordpress-two ᚴ:master 
❯ wp @vvv post list
+----+---------------+-------------+---------------------+-------------+
| ID | post_title    | post_name   | post_date           | post_status |
+----+---------------+-------------+---------------------+-------------+
| 1  | Hello world!2 | hello-world | 2019-10-24 23:17:18 | publish     |
+----+---------------+-------------+---------------------+-------------+
~/dev/vvv/www/wordpress-two ᚴ:master 
❯ wp @wordpress-two post list
+----+---------------+-------------+---------------------+-------------+
| ID | post_title    | post_name   | post_date           | post_status |
+----+---------------+-------------+---------------------+-------------+
| 1  | Hello world!2 | hello-world | 2019-10-24 23:17:18 | publish     |
+----+---------------+-------------+---------------------+-------------+
~/dev/vvv/www/wordpress-two ᚴ:master 
❯ 
```

I also suspect  that `ssh vvv.test` will fail for you with the same issue

**Comment by Mte90** [comment]
In the other pr I shared my terminal with that change

**Comment by tomjn** [comment]
I tried to reproduce your success and had none. `vagrant` is an unknown domain, this is not the solution, it's actually a significant regression.

**Comment by tomjn** [comment]
It also does not make sense. Why would SSH'ing into `vvv.test` require a password, but SSH'ing into the domain `vagrant` succeed? It's the same machine!

**Comment by Mte90** [comment]
Yeah this is something that I don't believe too but maybe on linux the behavior of vagrant is different?

**Comment by tomjn** [comment]
If it's adding a `vagrant` host then you can test for that with networking commands and `ssh vagrant`. `ssh vagrant@vagrant` should work. If you don't specify a username then it assumes it's the current user, e.g.:

```sh
❯ ssh vvv.test
tarendai@vvv.test's password: 
```

versus

```sh
❯ ssh vagrant@vvv.test
Last login: Tue Jan 12 17:21:49 2021 from 192.168.50.4
vagrant@vvv:~$ 
```

As for the virtual machine, a Linux VM is a Linux VM regardless of which OS the host is on. WP CLI isn't routing its SSH via the vagrant command, it's doing a direct connection to the OS.

So TLDR:

 - is `vagrant` listed in the hosts file?
 - is `.ssh/config` clear of anything that interferes?
 - where exactly does the Linux host think `vagrant` is located?
 - how does it know to use the `vagrant` user when connecting?
 - what happens if you directly SSH `ssh vagrant@vvv.test`?
 - what happens if you `ssh vagrant`?
 - what happens if you `ssh vagrant@vagrant`?

**Comment by tomjn** [comment]
Also @janw-me why didn't you raise an issue for this on the main VVV so we could track it? This issue doesn't show on any open systems as a result, and you're lucky we have a firehose channel in slack to show these things up :(

**Comment by janw-me** [comment]
@tomjn Sorry, I figured I'd comment where the if you'd like I can open an issue. But to answer your questions.

- is vagrant listed in the hosts file?
  No, that's not how it works
- is .ssh/config clear of anything that interferes?
  No, that not how it works
- where exactly does the Linux host think vagrant is located?
  it's not a host
- how does it know to use the vagrant user when connecting?
- what happens if you directly SSH ssh vagrant@vvv.test?
  Prompt password
- what happens if you ssh vagrant?
  `ssh: Could not resolve hostname vagrant: Temporary failure in name resolution`
  I never use this, I login with `vagrant ssh`
- what happens if you ssh vagrant@vagrant?
  `ssh: Could not resolve hostname vagrant: Temporary failure in name resolution`

`ssh: vagrant` in the ymal file does not use vagrant as a host. When I run `wp @vvv option get home --debug` In the debug output I see:
`Running SSH command: ssh -q -p 2222  -i /home/janw-wf/.vagrant.d/insecure_private_key -t 'vagrant@127.0.0.1' 'WP_CLI_RUNTIME_ALIAS='\''{"@vvv":{"path":"\/srv\/www\/test\/public_html"}}'\'' wp @vvv '\''option'\'' '\''get'\'' '\''home'\'' '\''--debug'\''' (0.038s)`

WP-cli detect `vagrant` and gets it data from `vagrant ssh-config`
Also see [the documentation](https://make.wordpress.org/cli/handbook/guides/running-commands-remotely/#using-an-ssh-connection):

>    The scheme argument defaults to ssh and alternately accepts options for vagrant, docker and docker-compose.

And the source code that does this magic [here](https://github.com/wp-cli/wp-cli/blob/b9b5d41c9c030b72bb597345084b0d6db4d1aac7/php/WP_CLI/Runner.php#L532) and [here](https://github.com/wp-cli/wp-cli/blob/1512cd07af5ffe0cdbcf563f368e182375618181/php/utils.php#L1013)

**Comment by tomjn** [comment]
VVV's IP on MacOS and windows is not `127.0.0.1`, and `ssh: vagrant` does not work on MacOS.

Likewise, `ssh vagrant@vvv.test` should not prompt you for a password, I cannot reproduce this on MacOS.

I suspect the actual problem is that the VM wants the vagrant insecure key. In which case, `vagrant ssh-config` is the solution.

Because setting the host to `vagrant` breaks MacOS, that change is a hard non-negotiable no from me. Unless it can be made to work, its introduction is a bug, not a fix.

**Comment by tomjn** [comment]
Also note that WP CLI's attempts to fallback to `vagrant ssh` will fail if the user tries to use the `local_dir` and `vm_dir` site config options

**Comment by tomjn** [comment]
Here's the `--debug` output here:

```
Debug (bootstrap): No readable global config found (0.169s)
Debug (bootstrap): Using project config: /Users/tarendai/dev/vvv/www/wordpress-two/wp-cli.yml (0.169s)
Debug (bootstrap): argv: /usr/local/bin/wp @test post list --debug (0.169s)
Debug (bootstrap): SSH scheme: vagrant (0.169s)
Debug (bootstrap): SSH user:  (0.169s)
Debug (bootstrap): SSH host:  (0.169s)
Debug (bootstrap): SSH port:  (0.169s)
Debug (bootstrap): SSH path:  (0.169s)
Debug (bootstrap): SSH key:  (0.169s)
Debug (bootstrap): Running SSH command: ssh -q -p 2222  -i /Users/tarendai/.vagrant.d/insecure_private_key -t 'vagrant@127.0.0.1' 'WP_CLI_RUNTIME_ALIAS='\''{"@test":{"path":"\/srv\/www\/wordpress-two\/public_html"}}'\'' wp @test '\''post'\'' '\''list'\'' '\''--debug'\''' (0.174s)
Error: Cannot connect over SSH using provided configuration.
```

**Comment by janw-me** [comment]
Weird, I don't have a Mac so I also can't reproduce it.
If it gives additional problems with custom configs it's better this way.
I'll just change is manually.

Thanks for looking into is this deep anyway :)

I don't see anything weird in the debug output.
If you want to see the ssh error inside you shoud try manually running in the terminal, Without the `-q` flag: 
```ssh -p 2222  -i /Users/tarendai/.vagrant.d/insecure_private_key -t 'vagrant@127.0.0.1' 'WP_CLI_RUNTIME_ALIAS='\''{"@test":{"path":"\/srv\/www\/wordpress-two\/public_html"}}'\'' wp @test '\''post'\'' '\''list'\'' '\''--debug'\'''```

In the past I've often seen that is gives a known hosts error with colleagues. When a vagrant was destroyed and recreated.

**Comment by tomjn** [comment]
So the problem is that because it SSH's to 127.0.0.1 the fingerprint does not match and SSH aborts. As a result, if you have ever SSH'd into localhost for any reason (e.g. loopbacks or other VMs), or if VVV is destroyed and recreated, the fingerprint will be different and WP CLI's work around will break.

I also don't see a mechanism to force the `vagrant ssh` fallback, which would solve this problem

**Comment by tomjn** [comment]
Likewise WP CLI could adjust its SSH options to skip host  key checking for `127.0.0.1`: `ssh -o StrictHostKeychecking=no hostname`

An issue over on WP CLI will be needed to get this fixed

**Comment by janw-me** [comment]
I'll look into that from the WP-cli side and file the issue.

But you can confirm you did see the error when manually running it without the `-q` error?

**Comment by tomjn** [comment]
I saw the fingerprinting issue when removing `-q`, I'm raising an issue now over at WP CLI

**Comment by tomjn** [comment]
https://github.com/wp-cli/wp-cli/issues/5488

**Comment by Mte90** [comment]
It is not clear to me, the error is happning with OSX as host for wpcli?

**Comment by tomjn** [comment]
Any machine that has 127.0.0.1 in its known hosts file already
